xdsl/parser.py

```diff
@@ -380,6 +380,26 @@ def parse_optional_string(self,
 
     T = TypeVar('T')
 
+    def parse_nested_list(self,
```

Comment:
Oh good point, this will fail if we parse a list of strings, since it will return a list of characters instead.
I'm wondering how we could write this while making it robust (i.e, this should work for every parsing functions)